scrapper.py

```diff
@@ -1,10 +1,140 @@
+                                         class_='a-card__header-left').find('a',
+                                                                 class_='a-card__title').text
+        # search start and end indexes of pattern like: "'digits' м"
+        indexes_of_square_number = re.search(r"\d+(\.\d*)?\sм", text_with_one_flat_square) 
+        square = float(text_with_one_flat_square[indexes_of_square_number.start():
+                                                        indexes_of_square_number.end()-1])
+        # parse offer id and url to offer
+        text_with_one_offer_id = each.find('div',
+                                     class_='a-card__header-left').find('a',
+                                                                     class_='a-card__title').get('href')
+        url_id = f'{URL}{text_with_one_offer_id}'
+        # search  start and end indexes of digits in str
+        indexes_of_offer_id = re.search(r"\d+", text_with_one_offer_id)
+        id = int(text_with_one_offer_id[indexes_of_offer_id.start():
+                                                    indexes_of_offer_id.end()])
+
+        data[id] = (url_id, price, square)
+        
+    return len(data)
+
+def get_number_of_options(soup: BeautifulSoup) -> int:
+    """ Function parse scrapped BeautifulSoup object and return number of options."""
+    
+    text_with_numbers_of_options = soup.find('div',
+                                     class_='a-search-subtitle search-results-nb').text.strip().split()
+    numbers_of_options = int(text_with_numbers_of_options[1]) # number of options
+
+    return numbers_of_options
+
+def get_cost_per_sq_m(data: dict) -> dict:
+    """ Function get dictinary with parsed data and return dictionary format {id : int(cost per m^2)}."""
+        
+    cost_per_sq_m = {key: int(data[key][1]/data[key][2]) for key in data.keys()}
+    
+    return cost_per_sq_m
+
+
+
+def main():
+    """ Function scrap site by URL which entered in command line.
+        Result: dictinary with parsed data in format {offer id : tuple(url offer id, flat price, flat square)}
+        and dictinary with cost per m^2 in format {id : int(cost per m^2)}.
+    """
+    
+# Use this argument for test e.g.: 
+# https://krisha.kz/prodazha/kvartiry/almaty-aujezovskij-mkr-dostyk/?das[live.rooms]=1
+# https://krisha.kz/prodazha/kvartiry/almaty-aujezovskij-mkr-zhetysu-1/?das[live.rooms]=1 
+
+    command_line_arguments = argv #parse argument from CLI
+    page_number = int(1)
+    numbers_of_added_options = int(0)
+    numbers_of_options = int(1)
+    data = {}
+     
+    accepted_url = check_cli_arg(command_line_arguments)
+
+    # Scrapped and parsed site pages while numbers of parsed offers less than
+    # numbers of offers stated by site
+    while numbers_of_added_options < numbers_of_options:
+
+        soup = pagescrapper(accepted_url, page_number)
+        numbers_of_options = get_number_of_options(soup)
```

Comment:
@Shuk0, it's maybe rarely if any user add or delete his offer to/from site during scrapper work. But this code not solve this problem. IMHO, there are two variants: 
 - get numbers of options only one time(condition with count in the loop).
 - wrap scrapper in function. If numbers of options are different try again. After 2 or 3 trying end with error message. (But it's difficult to test this feature)

Reply:
It's fair point that user may delete an offer while script is executing. However, I would suggest to do an exceptions handler that catches this type of issue. Let's imagine that user deleted an offer that was published on the second page. It means number of pages may change from X to X-1. I would suggest to do the handler that checks if page exists or if there is non-zero content on a page that it processed at this moment.

Reply:
In general, we have several options:
1. There are two or more offers may be added or deleted (Very rarely)
2. Offer added to almost scrapped part -> we don't write this offer and one offer be rewrite
3. Offer added to not scrapped part -> OK. we write all offers
4. Offer deleted from almost scrapped part -> we write old(deleted) offer and miss one offer which rebased to scrapped page instead first one
5. Offer deleted  from not scrapped part -> we write all offers. In current code we rewrite numbers of offers.

One problem which you wrote, if we scrapped penalt page: 
- in (2 or 3) options we'll not write one option on new page because we don't know about this page 
- in (4 or 5) options we'll try to scrapp empty page. It's not endless loop because we'll rewrite number of offers.

I think we have to add several check up:
1. if number of offers change we start over (max 2 times than error). (2-5) options checked and not same numbers of added and deleted offers (1)
2. if we try to rewrite last writed offer on previos page we start over (max 2 times than error). we check option (1) when users add to almost scrapped part and deleted from not scrapped part same numbers of offers.
3. Other options of (1) we can't check or it not affect to result